File: SI24004/types/dwRequest.ts
export interface Attachment {
    id: string;
    attachmentName: string;
    attachmentPath: string;
    attachmentType: string;
  }
  
  export interface Drawing {
    id: string;
    drawingName: string;
  }
  
  export interface Section {
    id: string;
    sectionName: string;
  }
  
  export interface Status {
    Id: string;
    StatusName: string;
    Ordinal: number;
  }
  
  export interface User {
    Id: string;
    UserName: string;
  }
  
  export interface DwRequestDTO {
    Id: string;
    DrawingCode: string;
    RequestCode: string;
    DrawingName: string;
    SectionId: string | null;
    DrawingTypeId: string | null;
    StatusId?: string | null;
    CreatedDate: string; // DateOnly เป็น string
    CreatedBy: string;
    UpdateDate?: string | null;
    UpdateBy?: string;
    DrawingDescription: string;
    UserId?: string | null;
    AttachmentId?: string | null;
    Active?: boolean;
    IsDelete?: boolean;
    Attachment?: Attachment[] | null;  // ใช้พหูพจน์สำหรับหลาย attachment
    DrawingType?: Drawing;
    Section?: Section;
    Status?: Status;
    User?: User;
  }
  
  
  

File: Backend/SI24004/Models/DwRequest.cs
using System;
using System.Collections.Generic;

namespace SI24004.Models;

public partial class DwRequest
{
    public Guid Id { get; set; }

    public string? DrawingCode { get; set; }

    public string? RequestCode { get; set; }

    public string? DrawingName { get; set; }

    public Guid SectionId { get; set; }

    public Guid DrawingTypeId { get; set; }

    public Guid StatusId { get; set; }

    public DateTime? CreatedDate { get; set; }

    public string? CreatedBy { get; set; }

    public DateTime? UpdateDate { get; set; }

    public string? UpdateBy { get; set; }

    public string? DrawingDescription { get; set; }

    public Guid? UserId { get; set; }

    public Guid? AttachmentId { get; set; }

    public bool Active { get; set; }

    public bool IsDelete { get; set; }

    public bool? DrawingRevise { get; set; }

    public virtual Attachment Attachment { get; set; }

    public virtual Drawing DrawingType { get; set; }

    public virtual Section Section { get; set; }

    public virtual Status Status { get; set; }

    public virtual User User { get; set; }
}
